### PR TITLE
feat(admin): Use dynamic filters from the API [EP-2767]

### PR DESCRIPTION
**Pull request by ancashoria**

### CODE REVIEW

File: packages/earth-admin/src/utils/hooks.tsx [hunk 1]
```diff
@@ -145,3 +149,28 @@ export function mergePages(pagedResponse: any[]): IMergedResults {
...
   );
 }
+
+/**
+ * Queries the API and returns dynamic filters grouped by their keys
+ */
+export function useFilters(resource: string, fetcher: (any) => Promise<any>, options: object = {}) {
+  const { selectedGroup } = useAuth0();
+  const query = {
+    select: 'name,slug',
```
Comment (corinacioloca) [inline packages/earth-admin/src/utils/hooks.tsx]: y u need name & slug?

File: packages/earth-admin/src/pages-client/places/new.tsx [hunk 2]
```diff
@@ -33,15 +33,17 @@ import { ContentLayout } from '@app/layouts';
...
   onDataChange?: () => {};
+  dynamicOptions?: {
+    type?: any[];
+  };
 }
 
 export function NewPlace(props: IProps) {
-  const { onDataChange = noop } = props;
+  const { onDataChange = noop, dynamicOptions } = props;
+  const { type: typeOptions = [] } = dynamicOptions;
```
Comment (corinacioloca) [inline packages/earth-admin/src/pages-client/places/new.tsx]: placeTypeOptions is more relevant

File: packages/earth-admin/src/pages-client/layers/new.tsx [hunk 3]
```diff
@@ -38,20 +38,27 @@ import { CUSTOM_STYLES, SELECT_THEME } from '@app/theme';
...
 export function NewLayer(props: IProps) {
-  const { onDataChange = noop } = props;
+  const { onDataChange = noop, dynamicOptions } = props;
   const { selectedGroup } = useAuth0();
-
   const { register, watch, formState, errors, setValue, control, handleSubmit } = useForm({
     mode: 'onChange',
   });
+  const {
+    category: categoryOptions = [],
```
Comment (corinacioloca) [inline packages/earth-admin/src/pages-client/layers/new.tsx]: layerCategoryOptions, layerTypeOptions, layerProviderOptions is more relevant

File: packages/earth-admin/src/utils/hooks.tsx (hunk 1 above)
Comment (ancashoria) [reply packages/earth-admin/src/utils/hooks.tsx]: I need a field that's present on every request. Definitely don't need both. I'll change it to 'id'

**Review by corinacioloca — APPROVED** [review]

**Pull request merged**